### ISSUE TITLE
2.2.0 Migration of manual Lucene indexes

### ISSUE THREAD

**Issue by antonzherdev** [orientechnologies/orientdb#6189]
## Expected behavior and actual behavior

I had manual Lucene indexes in 2.1.17. I updated to 2.2.0. When I open db it gives me the error for every index:
```
ERROR c.o.l.i.OLuceneFullTextIndex [lt-dispatcher-3] - $ANSI{green {db=db}} Error during load of index 'i13_1__
w13_1_Error__search'
java.lang.NullPointerException: null
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.loadExternalIndexEngine(OAbstractPaginatedStorage.java:1549)
        at com.orientechnologies.orient.core.index.OIndexAbstract.loadFromConfiguration(OIndexAbstract.java:309)
        at com.orientechnologies.orient.core.index.OIndexManagerShared.fromStream(OIndexManagerShared.java:414)
        at com.orientechnologies.orient.core.type.ODocumentWrapperNoClass.reload(ODocumentWrapperNoClass.java:74)
        at com.orientechnologies.orient.core.index.OIndexManagerAbstract.load(OIndexManagerAbstract.java:85)
        at com.orientechnologies.orient.core.index.OIndexManagerAbstract.load(OIndexManagerAbstract.java:57)
        at com.orientechnologies.orient.core.metadata.OMetadataDefault$2.call(OMetadataDefault.java:191)
        at com.orientechnologies.orient.core.metadata.OMetadataDefault$2.call(OMetadataDefault.java:181)
        at com.orientechnologies.common.concur.resource.OSharedContainerImpl.getResource(OSharedContainerImpl.java:58)
        at com.orientechnologies.orient.core.storage.OStorageAbstract.getResource(OStorageAbstract.java:149)
        at com.orientechnologies.orient.core.metadata.OMetadataDefault.init(OMetadataDefault.java:181)
        at com.orientechnologies.orient.core.metadata.OMetadataDefault.load(OMetadataDefault.java:89)
        at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.initAtFirstOpen(ODatabaseDocumentTx.java:3004)
        at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.open(ODatabaseDocumentTx.java:243)
        at com.erp.db.DbActor.preStart(DbActor.scala:51)
        at akka.actor.Actor$class.aroundPreStart(Actor.scala:487)
        at com.erp.db.DbActor.aroundPreStart(DbActor.scala:31)
        at akka.actor.ActorCell.create(ActorCell.scala:590)
        at akka.actor.ActorCell.invokeAll$1(ActorCell.scala:461)
        at akka.actor.ActorCell.systemInvoke(ActorCell.scala:483)
        at akka.dispatch.Mailbox.processAllSystemMessages(Mailbox.scala:282)
        at akka.dispatch.Mailbox.run(Mailbox.scala:223)
        at akka.dispatch.Mailbox.exec(Mailbox.scala:234)
        at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
        at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
        at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
        at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
```

If I try to access the index. For example to delete it, it gives me the error:
```
Engine with id -1 is not registered inside of storage
        Storage URL="plocal:/var/erpdb/db"
com.orientechnologies.orient.core.index.OIndexException: Engine with id -1 is not registered inside of storage
        Storage URL="plocal:/var/erpdb/db"
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.checkIndexId(OAbstractPaginatedStorage.java:1684)
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.doCallIndexEngine(OAbstractPaginatedStorage.java:1834)
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.callIndexEngine(OAbstractPaginatedStorage.java:1827)
        at com.orientechnologies.lucene.index.OLuceneIndexNotUnique.getTransactionChanges(OLuceneIndexNotUnique.java:403)
        at com.orientechnologies.lucene.index.OLuceneIndexNotUnique.put(OLuceneIndexNotUnique.java:118)
        at com.orientechnologies.lucene.index.OLuceneIndexNotUnique.put(OLuceneIndexNotUnique.java:44)
        at com.erp.db.WorkspaceSearch$ClassIndex.updateIndex(WorkspaceSearch.scala:66)
        at com.erp.db.WorkspaceSearch.updateIndex(WorkspaceSearch.scala:42)
        at com.erp.db.WorkspaceDb.save(WorkspaceDb.scala:49)
        at com.erp.db.WorkspaceDbUsers$$anonfun$4.apply(WorkspaceDbUsers.scala:89)
        at com.erp.db.WorkspaceDbUsers$$anonfun$4.apply(WorkspaceDbUsers.scala:82)
        at scala.collection.TraversableLike$WithFilter$$anonfun$map$2.apply(TraversableLike.scala:683)
        at scala.collection.immutable.HashMap$HashMap1.foreach(HashMap.scala:221)
        at scala.collection.immutable.HashMap$HashTrieMap.foreach(HashMap.scala:428)
        at scala.collection.TraversableLike$WithFilter.map(TraversableLike.scala:682)
        at com.erp.db.WorkspaceDbUsers.updateUserWorkspaces(WorkspaceDbUsers.scala:82)
        at com.erp.db.WorkspaceDbActor.com$erp$db$WorkspaceDbActor$$becomeActiveIfLoaded(WorkspaceDbActor.scala:75)
        at com.erp.db.WorkspaceDbActor$$anonfun$receive$1.applyOrElse(WorkspaceDbActor.scala:46)
        at akka.actor.Actor$class.aroundReceive(Actor.scala:482)
        at com.erp.db.WorkspaceDbActor.aroundReceive(WorkspaceDbActor.scala:21)
        at akka.actor.ActorCell.receiveMessage(ActorCell.scala:526)
        at akka.actor.ActorCell.invoke(ActorCell.scala:495)
        at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:257)
        at akka.dispatch.Mailbox.run(Mailbox.scala:224)
        at akka.dispatch.Mailbox.exec(Mailbox.scala:234)
        at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
        at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
        at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
        at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
```

So the main problem is that I cannot do anything about this. I can create new indexes but I cannot drop old. 

I expect that it should be able to convert indexes in a new format if required, even if it's manual indexes. Or at least there should be a way how to drop them.

## Steps to reproduce the problem

1. Create manual Lucene index in 2.1;
2. Start the database with 2.2.

## Important Questions

How to drop these broken indexes?

### Runninng Mode
- [X] Embedded, using PLOCAL access mode
- [ ] Embedded, using MEMORY access mode
- [ ] Remote

### Misc
- [ ] I have a distributed setup with multiple servers. How many? 
- [ ] I'm using the Enterprise Edition

### OrientDB Version
- [ ] v2.0.x - Please specify last number: 
- [ ] v2.1.x - Please specify last number: 
- [X] v2.2.x - Please specify last number: 0

### Operating System
- [X] Linux
- [X] MacOSX
- [ ] Windows
- [ ] Other Unix
- [ ] Other, name? 

### Java Version
- [ ] 6
- [ ] 7
- [X] 8

**Comment by robfrank** [comment]
This bug should be resolved and fix will be included in 2.2.1. In the meanwhile, can you test import from 2.1.x with latest snapshot: 
https://oss.sonatype.org/content/repositories/snapshots/com/orientechnologies/orientdb-community/2.2.1-SNAPSHOT/

**Comment by antonzherdev** [comment]
@robfrank Yes, it's working well for me. Thanks!
